RFC-0001-row-id.md

```diff
@@ -0,0 +1,80 @@
+# **RFC 1 for Presto**
+
+## Row ID
+
+Proposers
+
+* Brian Dahmen
+* Devang Shah
+* Elliotte Rusty Harold
+* Serge Druzkin
+* Sreeni Viswanadha
+
+## Related Issues
+
+* Row ID support #22078
+* Binary IDs for Hive partitions #22103
+* Binary rowID type #22053
+* Start adding row\_id hidden column to Hive #22008
+
+
+## Summary
+
+A row ID is a binary key for a row that is globally unique across the warehouse.
+It identifies a specific row at a specific version of a partition’s data. A row
+ID will continue to locate the same row in the warehouse until the underlying
+partition is mutated.
+
+## Background
+
+Queries can produce lists of row IDs that can be saved and used as input to
+future queries for quick selection. A list of row IDs can be used include or
+exclude rows to form a subset from a table.
+
+### Goals
+
+Row IDs resolve to the same value across compute engines. In this way they can
+be used as a strong-equality comparison across compute engines.
+
+### Non-goals
+
+Comparison and sorting of row IDs.
+
+## Proposed Implementation
+
+A row ID is a VARBINARY pseudo-column, also known as a hidden column. Each row
```

Comment:
Why does it need to be exposed as a column, instead of being encapsulated by the connector?

Reply:
In a SQL query it needs to appear as a column. I think you're only asking about what happens after the query `SELECT $row_id from table_name` begins moving through Presto. I'm going to defer to others to argue for this, but I seem to recall someone mentioning that making this a low level column improves the optimization and query planning. Meta-folk, am I remembering that right?

Reply:
OK, this doc is missing, what use does one have for $row_id, and how is it used for query optimization?  Why can't $row_id be connector-supplied?